src/main/java/org/candlepin/model/ContentOverrideCurator.java

```diff
@@ -42,12 +45,14 @@ public ContentOverrideCurator(Class<T> entityType, String parentAttrName) {
         this.parentAttr = parentAttrName;
     }
 
-    @SuppressWarnings("unchecked")
-    public CandlepinQuery<T> getList(Parent parent) {
-        DetachedCriteria criteria = DetachedCriteria.forClass(this.entityType())
-            .add(Restrictions.eq(parentAttr, parent));
+    public List<T> getList(Parent parent) {
+        CriteriaBuilder cb = this.getEntityManager().getCriteriaBuilder();
+        CriteriaQuery<T> query = cb.createQuery(this.entityType());
+        Root<T> root = query.from(this.entityType());
+        query.select(root);
+        query.where(cb.equal(root.get(parentAttr), parent));
 
-        return this.cpQueryFactory.<T>buildQuery(this.currentSession(), criteria);
+        return this.listByCriteria(query);
```

Comment:
Going through `listByCriteria` adds overhead we didn't have before... kind of (CPQ magic not withstanding). Since we're just a list (aptly matching our poor method name; side note: we need not maintain existing method names when they're not good), we need not be worried about the paging bits. If this method is used primarily for a paged endpoint, we should be updating this method to be more explicit about that fact rather than relying on magic.

Hopefully a simple JPA query works here.

Reply:
I'm confused on this. We definitely don't need use the `AbstractHibernateCurator.listByCriteria` method, but that method is only creating the query and then returning the result list:

```
return this.getEntityManager()
    .createQuery(query)
    .getResultList();
```

It is not doing anything with paging and I don't believe that the results from this method need to be paged. I updated the method to just create the query and then return the result list.

Reply:
My mistake. I confused it with the `listByCriteria` that also does paging. You can disregard this comment.